loopy/transform/iname.py

```diff
@@ -818,6 +815,45 @@ def map_instruction(self, kernel, insn):
                 for iname in insn.within_inames)
         return insn.copy(within_inames=new_fid)
 
+    def map_kernel(self, kernel):
+        new_insns = [
+                # While subst rules are not allowed in assignees, the mapper
+                # may perform tasks entirely unrelated to subst rules, so
+                # we must map assignees, too.
+                insn if not self.within(kernel, insn, ()) else
```

Comment:
This line is the only change from parent class' `map_kernel` function. This is the reason for the 2 failures. Not sure why it's failing.